### ISSUE TITLE
Display large numbers

### ISSUE THREAD

**Issue by nathalierun** [jmoenig/Snap--Build-Your-Own-Blocks#1832]
Hello everybody,

I have a little project, using a Ramanujan formula, to calculate some decimals of Pi.
I know that Snap! is able to calculate with large numbers. But I don't know how to access them.
How can I print more than 15 decimals ?

Thank you for any help.

**Comment by brianharvey** [comment]
If you go to File>Libraries, you can select the "infinite precision integer" library.  It doesn't quite do what you want; floating point (decimal) values are still limited to 15 significant digits.  But it does allow for exact _rational_ arithmetic, and the numerators and denominators are infinite precision.  I'm thinking you could build your own long division procedure that would generate a string of digits instead of an actual decimal number.  And of course you also have to deal with that √2 somehow.

So, long story short, this library can help but it won't trivially give you high precision decimal values.

**Comment by nathalierun** [comment]
Le 18/08/2017 à 09:27, Brian Harvey a écrit :
> it does allow for exact /rational/ arithmetic, and the numerators and 
> denominators are infinite precision.

I see. Thank you.

So, I have now to create my own long division procedure that would 
generate a string of digits instead of an actual decimal number. I have 
to think about it...
